asapdiscovery-workflows/asapdiscovery/workflows/prep_workflows/protein_prep.py

```diff
@@ -276,3 +322,139 @@ def protein_prep_workflow(inputs: ProteinPrepInputs):
+    selector = MCSSelector()
+    pairs = selector.select(
+        [Ligand.from_smiles(largest_compound, compound_name="query_compound")],
+        complexes,
+        n_select=10,  # if we run out of references we can either
+        # increase this number or start taking from e.g. the second-largest compound
+        # we can also use this point to build multi-ref FECs if we need to
+    )
+    reference_complexes = []
+    for pair in pairs:
+        target_name = pair.complex.target.target_name
+        logger.info(f"Found a match with {target_name}; attempting prep")
+        reference_complexes.append(target_name)
+        prepped_complex = prepper.prep(
+            inputs=[pair.complex],
+            use_dask=inputs.use_dask,
+            dask_client=dask_client,
+            cache_dir=inputs.cache_dir,
+        )[0]
+        logger.info(f"Checking that the prepped complex is compatible with OpenMM")
+        prepped_complex.target.to_pdb_file(
+            f"{target_name}.pdb"
+        )  # @HMO: this should be in a tmpdir?
+        # @HMO: can we the below more elegantly? I don't really like this setup with the below ValueError.
+        try:
+            prepped_simulation = create_protein_only_system(f"{target_name}.pdb")
+        except Exception as e:
+            prepped_simulation = False
+            logger.warn(
+                f"Reference {target_name} is not compatible with OpenMM:\n{e}\n\n ..trying next reference.."
+            )
+            pass
+        if prepped_simulation:
+            logger.info(f"{target_name} is compatible")
+            smiles_reference, smiles_query, similarity = get_similarity(
+                pair.complex.ligand.smiles, largest_compound
+            )
+            logger.info(
+                f"Using reference {target_name} such that\nReference compound: {smiles_reference}\nQuery compound from FECs set: {smiles_query}\nSimilarity: {similarity}"
+            )
+            if similarity < 0.2:
+                logger.warn(
+                    f"Low similarity with reference crystal pose ligand: {similarity}, check prep/docking results carefully"
+                )
+            return [prepped_complex]
+        else:
+            pass
+    raise ValueError(  # if we reach this point then we've failed to find a reference
+        f"No references were found for query ligand {largest_compound} in references: {reference_complexes}"
+    )
+
+
+#### BELOW SHOULD BE SOMEWHERE ELSE?
+import openmm
+from openff.toolkit import ForceField, Topology
+from openmm import unit as openmm_unit
+
+
+### SOMETHING GOING WRONG HERE - NO BOX?!
+def create_protein_only_system(pdb_path):
+    # attempt to make an OpenMM system with the prepped protein.
+    # follows https://docs.openforcefield.org/en/latest/examples/openforcefield/openff-toolkit/toolkit_showcase/toolkit_showcase.html
+    top = Topology.from_pdb(pdb_path)
+    sage_ff14sb = ForceField("openff-2.1.0.offxml", "ff14sb_off_impropers_0.0.3.offxml")
```

Comment:
@jthorton notes we should use the FFs we use in production forr `asap-alchemy`

Reply:
The default force fields can be found [here](https://github.com/OpenFreeEnergy/gufe/blob/29f98ec0227ed9f30ea6b84096dc13818a9ac8d3/gufe/settings/models.py#L161-L166). I think we just need to make an openMM system using the topology and the force fields listed and provided we don't hit an exception the pdb should be valid. Something like this should work 
```python
forcefield = openmm.ForceField([
        "amber/ff14SB.xml",  # ff14SB protein force field
        "amber/tip3p_standard.xml",  # TIP3P and recommended monovalent ion parameters
        "amber/tip3p_HFE_multivalent.xml",  # for divalent ions
        "amber/phosaa10.xml",  # Handles THE TPO
    ])
pdb = openmm.app.PDBFile(pdb_path)
try:
    _ = forcefield.createSystem(pdb.topology)
except Exception as e:
    raise InvalidStructureError from e
```